akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -36,6 +36,16 @@ import scala.concurrent.{ Future, Promise }
  */
 abstract class GraphStageWithMaterializedValue[+S <: Shape, +M] extends Graph[S, M] {
 
+  /**
+   * Grants eager access to materializer for special purposes.
+   *
+   * INTERNAL API
+   */
+  @InternalApi
+  private[akka] def createLogicAndEagerMaterializedValue(
+      inheritedAttributes: Attributes,
+      materializer: Materializer): (GraphStageLogic, M) = createLogicAndMaterializedValue(inheritedAttributes)
+
```

Comment:
My intent was to have it internal since it's only useful in very special cases (and probably better to provide an implementation to cover them) rather than opening it to everyone.

Reply:
Hmm, ok, yeah maybe you are right. Thinking some more about this it could perhaps also be a bit dangerous or at least different to use this materializer over the submaterializer available from the interpreter.

Reply:
We can always easily open it up later if needed

Reply:
I still think the method should be an overload though, the difference is that it gets a materializer as a parameter, not that it is more "eager" than the other factory method.